Decoration plugin optionally force decorations

allomanta opened the pull request:
This PR adds an option to the decoration plugin that ignores the clients request for client side decorations, see issue [2598](https://github.com/WayfireWM/wayfire/issues/2598).

ammen99 commented:
If we do this, maybe it should be a criteria/view_matcher option so that users can force decorations for particular views only? I think this will make the option useful for more people.

allomanta commented:
Good point, I hadn't considered that. I'm personally just using this to have consistent borders around all my windows ala hyprland or cosmic. I changed it to a view matcher and updated the description and variables to better match the ignore config option. The commit is a little messier as uncrustify seems to have formatted a bunch as well.

Also I wanted to say thanks for developing wayfire!

ammen99 commented:
I think the code style got messed up, please use uncrustify (we even currently use a fork due to upstream bugs)

allomanta commented:
Yeah it seems like some disagreement with my setup and running uncrustify messed up the code style. I have now undone that commit and added my changes back and I think the style should be correct now.

ammen99 reviewed (approved):
LGTM now, thanks for the PR!

allomanta commented:
Is there a way to see the error in the pipeline? Because running the command `git ls-files | grep "hpp$\|cpp$" | xargs uncrustify -c uncrustify.ini --no-backup` is not producing any changes in the 2 files here. Maybe I need to use that fork you mentioned?

ammen99 commented:
> Is there a way to see the error in the pipeline? Because running the command `git ls-files | grep "hpp$\|cpp$" | xargs uncrustify -c uncrustify.ini --no-backup` is not producing any changes in the 2 files here. Maybe I need to use that fork you mentioned?

Yes, you can, just take a look at the second-to-last command in the CI output, it prints the full diff: https://github.com/WayfireWM/wayfire/actions/runs/14062744211/job/39378080329?pr=2606#step:8:1

It seems that the error is tabs vs spaces.

allomanta commented:
Ah I was looking at the next CI step, but I think it was tabs vs spaces, should hopefully be fixed now.

ammen99 commented:
Looks good now :)
